File: Framework/CSharp/Framework/Framework/SmartContainer.cs
﻿/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
using System.Collections.Generic;

namespace Smartkernel.Framework
{
    /// <summary>
    /// 容器
    /// </summary>
    /// <typeparam name="TKey">键</typeparam>
    /// <typeparam name="TValue">值</typeparam>
    public static class SmartContainer<TKey, TValue>
    {
        private static object syncObject = new object();

        private static Dictionary<TKey, TValue> container;

        /// <summary>
        /// 静态构造函数
        /// </summary>
        static SmartContainer()
        {
            container = new Dictionary<TKey, TValue>();
        }

        /// <summary>
        /// 增加键值对
        /// </summary>
        /// <param name="key">键</param>
        /// <param name="value">值</param>
        public static void Add(TKey key, TValue value)
        {
            lock (syncObject)
            {
                container.Add(key, value);
            }
        }

        /// <summary>
        /// 是否包含
        /// </summary>
        /// <param name="key">键</param>
        /// <returns>结果</returns>
        public static bool ContainsKey(TKey key)
        {
            return container.ContainsKey(key);
        }

        /// <summary>
        /// 清空容器
        /// </summary>
        public static void Clear()
        {
            lock (syncObject)
            {
                container.Clear();
            }
        }

        /// <summary>
        /// 删除
        /// </summary>
        /// <param name="key">键</param>
        public static void Remove(TKey key)
        {
            lock (syncObject)
            {
                container.Remove(key);
            }
        }

        /// <summary>
        /// 获取
        /// </summary>
        /// <param name="key">键</param>
        /// <returns>值</returns>
        public static TValue Get(TKey key)
        {
            return container[key];
        }

        /// <summary>
        /// 设置
        /// </summary>
        /// <param name="key">键</param>
        /// <param name="value">值</param>
        public static void Set(TKey key, TValue value)
        {
            lock (syncObject)
            {
                container[key] = value;
            }
        }
    }
}

File: Framework/Java/framework/framework/smartkernel/framework/SmartContainer.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework;

import java.util.*;

/**
 * 容器
 * 
 */
public class SmartContainer {
	private static Object syncObject = new Object();

	private static HashMap<Object, Object> container;

	/**
	 * 静态构造函数
	 * 
	 */
	static {
		container = new HashMap<Object, Object>();
	}

	/**
	 * 增加键值对
	 * 
	 * @param key
	 *            ，键
	 * @param value
	 *            ，值
	 */
	public static <TKey, TValue> void add(TKey key, TValue value) {
		synchronized (syncObject) {
			container.put(key, value);
		}
	}

	/**
	 * 是否包含
	 * 
	 * @param key
	 *            ，键
	 * @return，结果
	 */
	public static <TKey> boolean containsKey(TKey key) {
		return container.containsKey(key);
	}

	/**
	 * 清空容器
	 */
	public static void clear() {
		synchronized (syncObject) {
			container.clear();
		}
	}

	/**
	 * 删除
	 * 
	 * @param key
	 *            ，键
	 */
	public static <TKey> void remove(TKey key) {
		synchronized (syncObject) {
			container.remove(key);
		}
	}

	/**
	 * 获取
	 * 
	 * @param key
	 *            ，键
	 * @return，值
	 */
	@SuppressWarnings("unchecked")
	public static <TKey, TValue> TValue get(TKey key) {
		return (TValue) container.get(key);
	}

	/**
	 * 设置
	 * 
	 * @param key，键
	 * @param value，值
	 */
	public static <TKey, TValue> void set(TKey key, TValue value) {
		synchronized (syncObject) {
			container.put(key, value);
		}
	}
}
